src/main/java/cubicchunks/server/PlayerCubeMap.java

```diff
@@ -697,52 +652,43 @@ void addToUpdateEntry(PlayerCubeMapEntry cubeWatcher) {
 		this.cubeWatchersToUpdate.add(cubeWatcher);
 	}
 
+	// CHECKED: 1.10.2-12.18.1.2092
 	void removeEntry(PlayerCubeMapEntry cubeWatcher) {
-		long address = cubeWatcher.getCubeAddress();
+		CubeCoords cubePos = cubeWatcher.getCubePos();
 		cubeWatcher.updateInhabitedTime();
-		this.cubeWatchers.remove(address);
+		this.cubeWatchers.remove(cubePos.getCubeX(), cubePos.getCubeY(), cubePos.getCubeZ());
 		this.cubeWatchersToUpdate.remove(cubeWatcher);
 		this.cubesToGenerate.remove(cubeWatcher);
 		this.cubesToSendToClients.remove(cubeWatcher);
+		if(cubeWatcher.getCube() != null) {
+			cubeWatcher.getCube().getTickets().remove(cubeWatcher); // remove the ticket, so this Cube can unload
+		}
 		//don't unload, ChunkGc unloads chunks
 	}
 
+	// CHECKED: 1.10.2-12.18.1.2092
 	public void removeEntry(PlayerCubeMapColumnEntry entry) {
 		ChunkPos pos = entry.getPos();
-		long address = getAddress(pos.chunkXPos, pos.chunkZPos);
 		entry.updateChunkInhabitedTime();
-		this.columnWatchers.remove(address);
+		this.columnWatchers.remove(pos.chunkXPos, pos.chunkZPos);
+		this.columnsToGenerate.remove(pos);
+		this.columnsToSendToClients.remove(pos);
 	}
 
 	public PlayerCubeMapEntry getCubeWatcher(CubeCoords pos) {
-		return getCubeWatcher(pos.getAddress());
+		return this.cubeWatchers.get(pos.getCubeX(), pos.getCubeY(), pos.getCubeZ());
 	}
 
-	public PlayerCubeMapEntry getCubeWatcher(int cubeX, int cubeY, int cubeZ) {
-		return getCubeWatcher(AddressTools.getAddress(cubeX, cubeY, cubeZ));
-	}
-
-	private PlayerCubeMapEntry getCubeWatcher(long address) {
-		return this.cubeWatchers.get(address);
-	}
-
-	public PlayerCubeMapColumnEntry getColumnWatcher(long address) {
-		return this.columnWatchers.get(address);
-	}
-
-	private boolean cubeExists(long address) {
-		int cubeX = getX(address);
-		int cubeY = getY(address);
-		int cubeZ = getZ(address);
-		return cubeCache.getLoadedCube(cubeX, cubeY, cubeZ) != null;
+	public PlayerCubeMapColumnEntry getColumnWatcher(ChunkPos pos) {
+		return this.columnWatchers.get(pos.chunkXPos, pos.chunkZPos);
 	}
 
 	public ICubicWorldServer getWorld() {
 		return (ICubicWorldServer) this.getWorldServer();
 	}
 
 	public boolean contains(CubeCoords coords) {
-		return this.cubeWatchers.containsKey(coords.getAddress());
+		return this.cubeWatchers.get(coords.getCubeX(), coords.getCubeY(), coords.getCubeZ()) != null;
```

Comment:
You can only put the value directly, but using get with other addressable would be nice feature to have in X(Y)ZMap